core/modules/config.go

```diff
@@ -42,6 +43,18 @@ type ModuleConfig struct {
 	Codegen *ModuleCodegenConfig `json:"codegen,omitempty"`
 }
 
+type ModuleConfigUserFields struct {
+	// The self-describing json $schema
+	Schema string `json:"$schema,omitempty"`
+}
+
+// ModuleConfigWithUserFields is the config for a single module as loaded from a dagger.json file.
+// Includes additional fields that should only be set by the user.
+type ModuleConfigWithUserFields struct {
```

Comment:
Took a swing at separating these "user fields" from the dagger-configurable fields.

Needs to be set up this way based on https://stackoverflow.com/questions/52294027/golang-structure-doesnt-unmarshal-properly-when-using-a-custom-unmarshal-for-a/52305457#52305457